File: twoPointers/maxArea.py
def maxArea(height):
    l = 0
    r = len(height) - 1
    area = (r-l) * min(height[l], height[r])
    while l < r:
        if height[l] < height[r]:
            l += 1
        else:
            r -= 1
        area = max(area, (r-l)*min(height[l], height[r]))
    return area

# https://leetcode.com/problems/container-with-most-water/

File: twoPointers/maxArea.js
function maxArea(height) {
  let left = 0;
  let right = height.length - 1;
  let area = 0;

  while (left < right) {
    area = Math.max(area, (right - 1) * Math.min(height[left], height[right]));
    if (height[left] < height[right]) left++;
    else {
      right--;
    }
  }
  return area;
}

console.log(maxArea([1, 8, 6, 2, 5, 4, 8, 3, 7]));
